### ISSUE TITLE
Boostd v1.5.1-rc2 randomly causes server to crash

### ISSUE THREAD

**Issue by stuberman** [filecoin-project/boost#1011]
### Checklist

- [X] This is **not** a question or a support request. If you have any boost related questions, please ask in the [discussion forum](https://github.com/filecoin-project/boost/discussions).
- [X] This is **not** a new feature or enhancement request. If it is, please open a [new idea discussion](https://github.com/filecoin-project/boost/discussions/new?category=ideas) instead. New feature and enhancement requests would be entertained by the boost team after a thorough discussion only.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/boost/issues) and the [discussion forum](https://github.com/filecoin-project/boost/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/boost/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to boost.

### Boost component

- [x] boost daemon - storage providers
- [ ] boost client
- [ ] boost UI
- [ ] boost data-transfer
- [ ] boost index-provider
- [ ] Other

### Boost Version

```text
boostd version 1.5.1-rc2+git.2c5604e

Lotus v1.19.0-rc2

Ubuntu 20.04.5 LTS (GNU/Linux 5.15.0-53-generic x86_64)
```

### Describe the Bug

Running boostd causes server to randomly crash.
I start boostd with this (no other processes running)
`nohup boostd -vv run --pprof > /market/logs/boost.log 2>&1 &`

Attached is dump.out (with .txt added so GitHub will not reject

[dump.output.txt](https://github.com/filecoin-project/boost/files/10099496/dump.output.txt)

### Logging Information

```text
head -50 /market/logs/boost.log

2022-11-27T23:55:21.133Z	DEBUG	boostd	boostd/run.go:54	Checking full node version
2022-11-27T23:55:21.134Z	DEBUG	boostd	boostd/run.go:61	Remote full node version	{"version": "1.19.0-rc2+mainnet+git.2520b1644+api2.3.0"}
2022-11-27T23:55:21.134Z	DEBUG	boostd	boostd/run.go:67	Checking full node sync status
Worker: 8556; Base: 0; Target: 0 (diff: 0)
State: idle; Current Epoch: 0; Todo: 0

Done!
2022-11-27T23:55:21.144Z	DEBUG	boostd	boostd/run.go:91	Instantiating new boost node
2022-11-27T23:55:21.257Z	INFO	modules	modules/rcmgr.go:42	libp2p resource manager is enabled
2022-11-27T23:55:21.257Z	INFO	modules	modules/rcmgr.go:86	adjusted default resource manager limits
2022-11-27T23:55:21.257Z	DEBUG	rcmgr	resource-manager/limit.go:84	initializing new limiter with config	{"limits": {"System":{"Streams":32768,"StreamsInbound":8192,"StreamsOutbound":32768,"Conns":2048,"ConnsInbound":1024,"ConnsOutbound":1024,"FD":524288,"Memory":4294967296},"Transient":{"Streams":2264,"StreamsInbound":1132,"StreamsOutbound":2264,"Conns":315,"ConnsInbound":157,"ConnsOutbound":315,"FD":131072,"Memory":1086324736},"AllowlistedSystem":{"Streams":18112,"StreamsInbound":9056,"StreamsOutbound":18112,"Conns":1132,"ConnsInbound":566,"ConnsOutbound":1132,"FD":524288,"Memory":8556380160},"AllowlistedTransient":{"Streams":2264,"StreamsInbound":1132,"StreamsOutbound":2264,"Conns":315,"ConnsInbound":157,"ConnsOutbound":315,"FD":131072,"Memory":1086324736},"ServiceDefault":{"Streams":16384,"StreamsInbound":4096,"StreamsOutbound":16384,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1119879168},"Service":{"libp2p.autonat":{"Streams":95,"StreamsInbound":95,"StreamsOutbound":95,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":20643840},"libp2p.holepunch":{"Streams":189,"StreamsInbound":94,"StreamsOutbound":94,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":37093376},"libp2p.identify":{"Streams":1132,"StreamsInbound":566,"StreamsOutbound":566,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":37093376},"libp2p.ping":{"Streams":566,"StreamsInbound":566,"StreamsOutbound":566,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":37093376},"libp2p.relay/v1":{"Streams":2264,"StreamsInbound":2264,"StreamsOutbound":2264,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":148373504},"libp2p.relay/v2":{"Streams":2264,"StreamsInbound":2264,"StreamsOutbound":2264,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":148373504}},"ServicePeerDefault":{"Streams":318,"StreamsInbound":159,"StreamsOutbound":318,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":49676288},"ServicePeer":{"libp2p.autonat":{"Streams":2,"StreamsInbound":2,"StreamsOutbound":2,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"libp2p.holepunch":{"Streams":2,"StreamsInbound":2,"StreamsOutbound":2,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"libp2p.identify":{"Streams":32,"StreamsInbound":16,"StreamsOutbound":16,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"libp2p.ping":{"Streams":4,"StreamsInbound":2,"StreamsOutbound":3,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":8590458880},"libp2p.relay/v1":{"Streams":64,"StreamsInbound":64,"StreamsOutbound":64,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"libp2p.relay/v2":{"Streams":64,"StreamsInbound":64,"StreamsOutbound":64,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576}},"ProtocolDefault":{"Streams":16384,"StreamsInbound":4096,"StreamsOutbound":16384,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1415970816},"Protocol":{"/ipfs/id/1.0.0":{"Streams":1132,"StreamsInbound":566,"StreamsOutbound":566,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":37093376},"/ipfs/id/push/1.0.0":{"Streams":1132,"StreamsInbound":566,"StreamsOutbound":566,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":37093376},"/ipfs/ping/1.0.0":{"Streams":566,"StreamsInbound":566,"StreamsOutbound":566,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":37093376},"/libp2p/autonat/1.0.0":{"Streams":95,"StreamsInbound":95,"StreamsOutbound":95,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":20643840},"/libp2p/circuit/relay/0.1.0":{"Streams":5660,"StreamsInbound":5660,"StreamsOutbound":5660,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":148373504},"/libp2p/circuit/relay/0.2.0/hop":{"Streams":5660,"StreamsInbound":5660,"StreamsOutbound":5660,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":148373504},"/libp2p/circuit/relay/0.2.0/stop":{"Streams":5660,"StreamsInbound":5660,"StreamsOutbound":5660,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":148373504},"/libp2p/dcutr":{"Streams":189,"StreamsInbound":94,"StreamsOutbound":94,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":37093376},"/p2p/id/delta/1.0.0":{"Streams":1132,"StreamsInbound":566,"StreamsOutbound":566,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":37093376}},"ProtocolPeerDefault":{"Streams":381,"StreamsInbound":95,"StreamsOutbound":190,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":16777247},"ProtocolPeer":{"/ipfs/id/1.0.0":{"Streams":32,"StreamsInbound":16,"StreamsOutbound":16,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":8590458880},"/ipfs/id/push/1.0.0":{"Streams":32,"StreamsInbound":16,"StreamsOutbound":16,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":8590458880},"/ipfs/ping/1.0.0":{"Streams":4,"StreamsInbound":2,"StreamsOutbound":3,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":8590458880},"/libp2p/autonat/1.0.0":{"Streams":2,"StreamsInbound":2,"StreamsOutbound":2,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"/libp2p/circuit/relay/0.1.0":{"Streams":128,"StreamsInbound":128,"StreamsOutbound":128,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":33554432},"/libp2p/circuit/relay/0.2.0/hop":{"Streams":128,"StreamsInbound":128,"StreamsOutbound":128,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":33554432},"/libp2p/circuit/relay/0.2.0/stop":{"Streams":128,"StreamsInbound":128,"StreamsOutbound":128,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":33554432},"/libp2p/dcutr":{"Streams":2,"StreamsInbound":2,"StreamsOutbound":2,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"/p2p/id/delta/1.0.0":{"Streams":32,"StreamsInbound":16,"StreamsOutbound":16,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":8590458880}},"PeerDefault":{"Streams":2520,"StreamsInbound":1260,"StreamsOutbound":2520,"Conns":8,"ConnsInbound":4,"ConnsOutbound":8,"FD":8192,"Memory":1119879168},"Conn":{"Streams":0,"StreamsInbound":0,"StreamsOutbound":0,"Conns":1,"ConnsInbound":1,"ConnsOutbound":1,"FD":1,"Memory":1048576},"Stream":{"Streams":1,"StreamsInbound":1,"StreamsOutbound":1,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":16777216}}}
2022-11-27T23:55:21.266Z	INFO	modules	modules/storageminer_svc.go:67	Connecting storage service to miner
2022-11-27T23:55:21.266Z	INFO	modules	modules/storageminer_svc.go:30	Checking (svc) api version of ws://192.168.1.95:2345/rpc/v0
2022-11-27T23:55:21.325Z	INFO	modules	modules/storageminer_idxprov.go:90	Instantiated index provider engine	{"idxProvEnabled": true, "pid": "12D3KooWSsaFCtzDJUEhLQYDdwoFtdCMqqfk562UMvccFz12kYxU", "topic": "/indexer/ingest/mainnet", "retAddrs": ["/ip4/108.89.91.150/tcp/46717"], "extraGossipData": "f01278", "publisher": "data-transfer"}
2022-11-27T23:55:21.325Z	INFO	modules	modules/storageminer_svc.go:60	Connecting sealing service to miner
2022-11-27T23:55:21.325Z	INFO	modules	modules/storageminer_svc.go:30	Checking (svc) api version of ws://192.168.1.95:2345/rpc/v0
2022-11-27T23:55:21.551Z	INFO	modules	modules/storageminer.go:336	starting legacy storage provider
2022-11-27T23:55:21.576Z	INFO	modules	modules/storageminer_idxprov.go:100	Started index provider engine
2022-11-27T23:55:21.581Z	DEBUG	dagstore	dagstore@v0.5.5/dagstore.go:576	restored shard state on dagstore startup	{"shard": "baga6ea4seaqa22t3flekcete234uammlb6cm7np6tccyyakynj6ka2ggnur3mby", "shard state": "ShardStateAvailable", "shard error": null, "shard lazy": false}
2022-11-27T23:55:21.582Z	DEBUG	dagstore	dagstore@v0.5.5/dagstore.go:576	restored shard state on dagstore startup	{"shard": "baga6ea4seaqa26gj5gbw3nbhxbecu5o2d22f5p4z4dswuib26abaf2s7k7hk2jy", "shard state": "ShardStateErrored", "shard error": "failed to acquire reader of mount on initialization: failed to unseal deal 1734978: failed to unseal piece from sector 1045: unsealing piece: worker UnsealPiece call: storage call error 0: %!w(unsealing sector: acquire unsealed sector path (existing): allocate local sector for fetching: couldn't find a suitable path for a sector [Hostname: sealer])", "shard lazy": true}
```

### Repo Steps

1. Run '...'
2. Do '...'
3. See error '...'
...

**Comment by LexLuthr** [comment]
@stuberman Is the machine itself crashing or the boostd process? If it is the machine, can you check the syslogs and get us the relevant part for troubleshooting?

**Comment by stuberman** [comment]
The machine would crash randomly, this was after destroying a lot of dagstore shards that were pointing to sectors that no longer exist.

I have updated boost now to `boostd version 1.5.1-rc2+git.b5768c8` and will monitor to see if this still happens.

**Comment by dirkmc** [comment]
@stuberman can you give us any kind of logs that mind give us a clue? Do you think it could be a memory issue? Anything in syslog? Otherwise it's hard to know where to start.

**Comment by stuberman** [comment]
I do not see any information in my lotus/boost or Ubuntu logs that provide a clue. I did provide a pprof output link above if it helps.

More importantly, I hear that this is also a common issue at PiKNiK now that we converted most of our market nodes to Boost nodes as well as hearing this from other members of the community, which tells me it is not a simple hardware issue. I have also asked PiKNiK engineering staff to post to this thread as well with their comments and thoughts.

It might be worthwhile to broadly ask for those using boost in the community to report their experience.
